tests/test_batch_emitter.rs

```diff
@@ -74,3 +80,45 @@ async fn flush_emitter() {
 
     assert_eq!(350, all_events["good"]);
 }
+
+#[tokio::test]
+async fn successful_send_after_retry() {
+    let docker = Cli::default();
+    let (_container, micro_url) = setup(&docker);
+
+    let event_store = InMemoryEventStore::new(2, 1);
+
+    let counter = Arc::new(AtomicUsize::new(0));
```

Comment:
Nit: did you mean to add an assert for the counter here too? To check that it was actually incremented and requests had to be retried.